pkg/controller/volume/attachdetach/cache/actual_state_of_world.go

```diff
@@ -337,37 +322,86 @@ func (asw *actualStateOfWorld) SetVolumeMountedByNode(
 
 	nodeObj.mountedByNode = mounted
 	volumeObj.nodesAttachedTo[nodeName] = nodeObj
-
+	glog.V(4).Infof("SetVolumeMountedByNode volume %v to the node %q mounted %q",
+		volumeName,
+		nodeName,
+		mounted)
 	return nil
 }
 
-func (asw *actualStateOfWorld) MarkDesireToDetach(
-	volumeName api.UniqueVolumeName, nodeName string) (time.Duration, error) {
+func (asw *actualStateOfWorld) ResetDetachRequestTime(
+	volumeName api.UniqueVolumeName, nodeName string) {
 	asw.Lock()
 	defer asw.Unlock()
 
-	volumeObj, volumeExists := asw.attachedVolumes[volumeName]
-	if !volumeExists {
-		return time.Millisecond * 0, fmt.Errorf(
-			"failed to MarkDesireToDetach(volumeName=%v, nodeName=%q) volumeName does not exist",
-			volumeName,
-			nodeName)
+	volumeObj, nodeObj, err := asw.getNodeAndVolume(volumeName, nodeName)
+	if err != nil {
+		glog.Errorf("Failed to SetVolumeMountedByNode with error: %v", err)
```

Comment:
I believe this message is wrong (should it say ResetDetachRequestTime), and do we want to continue if we hit an error (I suspect we'll panic when we write to the nodesAttachedTo map) ?